### PR TITLE
feat: Introduce new status API for logs and traces

### PR DESCRIPTION
**Pull request by shorim**
## Description

Changes proposed in this pull request (what was done and why):

- Add new condition types `GatewayHealthy` and `ConfigurationGenerated` to the `TracePipeline` status
- Add new condition types `AgentHealthy` and `ConfigurationGenerated` to the `LogPipeline` status
- Add new condition type `AgentHealthy` to the `LogParser` status
- For `TracePipeline`, `LogPipeline` and `LogParser`, ensure that the old condition types `(Running` and `Pending`) still exist and are always appended to the end of the conditions list
- Add a deprecation note in the `Message` field for the old condition types `(Running` and `Pending`)
- Adjust the `printcolumn` to display only the new conditions for all pipelines
- Set the Telemetry CR status based only on the new conditions for the pipelines
- For the Telemetry CR, to have consistency with the `MetricComponentsHealthy` condition:
  - Change the reason when `LogComponentsHealthy` condition is `True` from `FluentBitDaemonSetReady` to `LogComponentsRunning `
  - Change the reason when `TraceComponentsHealthy` condition is `True` from `TraceGatewayDeploymentReady` to `TraceComponentsRunning`
- Add documentation for the new conditions and add a deprecation note in the docs for the  old condition types `(Running` and `Pending`)
- Update Busola extensions to include new conditions
- Adjust verifiers in e2e tests
- Add unit tests

Changes refer to particular issues, PRs or documents:

- #805 

## Traceability
- [x] The PR is linked to a GitHub issue.
- [x] New features have a milestone set.
- [x] New features have defined acceptance criteria in a corresponding GitHub Issue, and all criteria are satisfied with this PR.
- [x] The corresponding GitHub issue has a respective `area` and `kind` label.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.
- [x] Adjusted the documentation if the change is user-facing.
- [x] The feature is unit-tested
- [x] The feature is e2e-tested

### CODE REVIEW

File: docs/user/resources/01-telemetry.md [hunk 1]
```diff
@@ -105,42 +106,41 @@ The `state` attribute of the Telemetry CR is derived from the combined state of
...
-| True             | FluentBitDaemonSetReady    | Fluent Bit DaemonSet is ready                   |
-| False            | ReferencedSecretMissing    | One or more referenced Secrets are missing      |
-| False            | FluentBitDaemonSetNotReady | Fluent Bit DaemonSet is not ready               |
-| False            | ResourceBlocksDeletion     | The deletion of the module is blocked. To unblock the deletion, delete the following resources: LogPipelines (resource-1, resource-2,...), LogParsers (resource-1, resource-2,...) |
-| False            | UnsupportedLokiOutput | The grafana-loki output is not supported anymore. For integration with a custom Loki installation, use the `custom` output and follow [Installing a custom Loki stack in Kyma](https://github.com/kyma-project/examples/tree/main/loki).              |
+| Condition Status | Condition Reason           | Condition Message                                                                                                                                                                                                                        |
```
Comment (NHingerl) [inline docs/user/resources/01-telemetry.md:109]: Title case for headings, excellent!

Comment (shorim) [reply docs/user/resources/01-telemetry.md]: Thank you! 😄

File: docs/user/resources/02-logpipeline.md [hunk 2]
```diff
@@ -47,11 +48,30 @@ spec:
...
+    type: AgentHealthy
+  - lastTransitionTime: "2024-02-28T22:48:11Z"
+    message: ""
+    observedGeneration: 2
+    reason: ConfigurationGenerated
+    status: "True"
+    type: ConfigurationGenerated
+  - lastTransitionTime: "2024-02-28T22:48:24Z"
+    message: '[NOTE: The "Pending" type is deprecated] Fluent Bit DaemonSet is not
+      ready'
```
[the remark this answers is not in the document]
Comment (shorim) [reply docs/user/resources/02-logpipeline.md]: Good catch! I fixed it

**Review by NHingerl — APPROVED** [review]

**Review by chrkl — APPROVED** [review]

File: config/manager/kustomization.yaml [hunk 3]
```diff
@@ -9,3 +9,4 @@ images:
 - name: controller
   newName: europe-docker.pkg.dev/kyma-project/prod/telemetry-manager
   newTag: main
+  newTag: v20
```
[the remark this answers is not in the document]
Comment (shorim) [reply config/manager/kustomization.yaml:12]: Done in #852

File: docs/user/resources/01-telemetry.md [hunk 4]
```diff
@@ -16,6 +16,7 @@ kind: Telemetry
 metadata:
   name: default
   namespace: kyma-system
+  generation: 2
```
[the remark this answers is not in the document]
Comment (shorim) [reply docs/user/resources/01-telemetry.md:19]: We discussed that this is actually the current behaviour. The `metadata.generation` and the `observedGeneration` in conditions have the same value.

File: internal/conditions/conditions.go [hunk 5]
```diff
@@ -10,84 +10,113 @@ import (
...
+		Message:            CommonMessageFor(reason, messageMap),
 		ObservedGeneration: generation,
 	}
 }
 
 // CommonMessageFor returns a human-readable message corresponding to a given reason.
 // In more advanced scenarios, you may craft custom messages tailored to specific use cases.
-func CommonMessageFor(reason string) string {
-	if condMessage, found := message[reason]; found {
+func CommonMessageFor(reason string, messageMap map[string]string) string {
```
[the remark this answers is not in the document]
Comment (shorim) [reply internal/conditions/conditions.go:99]: Improved in #852

File: internal/reconciler/telemetry/log_components_checker.go [hunk 6]
```diff
@@ -21,18 +21,36 @@ func (l *logComponentsChecker) Check(ctx context.Context, telemetryInDeletion bo
...
 
 	var logParsers telemetryv1alpha1.LogParserList
 	err = l.client.List(ctx, &logParsers)
 	if err != nil {
-		return &metav1.Condition{}, fmt.Errorf("failed to list log parsers: %w", err)
+		return &metav1.Condition{}, fmt.Errorf("failed to get list of LogParsers: %w", err)
 	}
 
+	// Remove "Running" and "Pending" conditions, since the Telemetry status shouldn't depend on these deprecated conditions
+	l.removePendingAndRunningConditions(logPipelines.Items)
```
[the remark this answers is not in the document]
Comment (shorim) [reply internal/reconciler/telemetry/log_components_checker.go:34]: Done in #852

File: config/busola/logpipeline_busola_extension_cm.yaml [hunk 7]
```diff
@@ -26,10 +26,28 @@ data:
             - 'false'
           negative:
             - 'true'
-      - source: status.conditions[-1].type
-        name: Status
-        widget: Badge
     body:
+      - source: status.conditions
```
[the remark this answers is not in the document]
Comment (shorim) [reply config/busola/logpipeline_busola_extension_cm.yaml:30]: The removal of deprecated status conditions from Kyma dashboard is done in #854

File: internal/reconciler/telemetry/log_components_checker.go [hunk 8]
```diff
@@ -21,18 +21,36 @@ func (l *logComponentsChecker) Check(ctx context.Context, telemetryInDeletion bo
...
 	reason := l.determineReason(logPipelines.Items, logParsers.Items, telemetryInDeletion)
+	status := l.determineConditionStatus(reason)
```
[the remark this answers is not in the document]
Comment (shorim) [reply internal/reconciler/telemetry/log_components_checker.go:36]: We agreed to keep these helper functions for now as methods of the `logComponentsChecker `, `traceComponentsChecker ` and `metricComponentsChecker `. The reason behind this is that all of these helpers are in the same package. Thus, if we don't make them methods of the checkers, we will need to give them unique long names, like `determineLogComponentsConditionReason` , `determineLogComponentsConditionStatus` , `createLogComponentsConditionMessageForReason` , `addLogComponentsConditionReasonPrefix`

**Pull request merged**